src/Microsoft.VisualStudio.AppDesigner/ApplicationDesigner/ApplicationDesignerPanel.vb

```diff
@@ -275,6 +275,13 @@ Namespace Microsoft.VisualStudio.Editors.ApplicationDesigner
 
                         If DocData IsNot Nothing Then
                             ExistingDocDataPtr = Marshal.GetIUnknownForObject(DocData)
+                        Else
```

Comment:
The way I have this pictured in my mind; our editor factory gets called with the existing doc data - why do we have to look it up again? We should already be passed it.

Reply:
ie What does `DocData` represent if not the existing doc data?

Reply:
Your picture is correct, our editor factory does get called with the existing doc data. This method is the thing that calls it, via `OpenSpecificEditor`, and supplies that doc data.

The `DocData` property is the existing doc data that this instance of the ApplicationDesignerPanel knows about, and would only be a factor if `CreateDesigner()` is more than once on an instance. I didn't see that happen during debugging.